libbeat/common/file/interval_rotator.go

```diff
@@ -61,38 +62,31 @@ func newIntervalRotator(interval time.Duration) (*intervalRotator, error) {
 func (r *intervalRotator) initialize() error {
 	r.clock = realClock{}
 
-	switch r.interval {
-	case time.Second:
+	if r.interval < time.Minute {
```

Comment:
nit: better use switch statements instead of if-elseif-else... 

```
switch {
case r.interval < time.Minute:
   ...
case r.interval < time.Hour:
   ...
...
default:
    ...
}
```

How about defining some constants like `const day = 24 * time.Hour`?

Moving this selection into an extra function makes it testable (often it's the most simplest code that contains the bug :) ).